src/uu/lscpu/src/lscpu.rs

```diff
@@ -49,41 +61,145 @@ impl CpuInfos {
+
+    if let Some(addr_sizes) = find_cpuinfo_value(&contents, "address sizes") {
+        arch_info.add_child(CpuInfo::new("Address sizes", &addr_sizes, None))
     }
 
-    if json {
-        println!("{}", cpu_infos.to_json());
-    } else {
-        for elt in cpu_infos.lscpu {
-            println!("{}: {}", elt.field, elt.data);
+    if let Ok(byte_order) = fs::read_to_string("/sys/kernel/cpu_byteorder") {
+        match byte_order.trim() {
+            "big" => arch_info.add_child(CpuInfo::new("Byte Order", "Big Endian", None)),
+            "little" => arch_info.add_child(CpuInfo::new("Byte Order", "Little Endian", None)),
+            _ => eprintln!("Unrecognised Byte Order: {}", byte_order),
         }
     }
+
+    cpu_infos.push(arch_info);
+
+    // TODO: This is currently quite verbose and doesn't strictly respect the hierarchy of `/proc/cpuinfo` contents
+    // ie. the file might contain multiple sections, each with their own vendor_id/model name etc. but right now
+    // we're just taking whatever our regex matches first and using that
+    if let Some(vendor) = find_cpuinfo_value(&contents, "vendor_id") {
+        let mut vendor_info = CpuInfo::new("Vendor ID", &vendor, None);
+
+        if let Some(model_name) = find_cpuinfo_value(&contents, "model name") {
+            let mut model_name_info = CpuInfo::new("Model name", &model_name, None);
+
+            if let Some(family) = find_cpuinfo_value(&contents, "cpu family") {
+                model_name_info.add_child(CpuInfo::new("CPU Family", &family, None));
+            }
+
+            if let Some(model) = find_cpuinfo_value(&contents, "model") {
+                model_name_info.add_child(CpuInfo::new("Model", &model, None));
+            }
+
+            vendor_info.add_child(model_name_info);
+        }
+        cpu_infos.push(vendor_info);
+    }
+
+    print_output(cpu_infos, output_opts);
+
     Ok(())
 }
 
+fn print_output(infos: CpuInfos, out_opts: OutputOptions) {
+    if out_opts.json {
+        println!("{}", infos.to_json());
+        return;
+    }
+
+    fn indentation(depth: usize) -> usize {
+        // Indentation is 2 spaces per level, used in a few places, hence its own helper function
+        depth * 2
+    }
+
+    // Recurses down the tree of entries and find the one with the "widest" field name (taking into account tree depth)
+    fn get_max_field_width(info: &CpuInfo, depth: usize) -> usize {
+        let max_child_width = info
+            .children
+            .iter()
+            .map(|entry| get_max_field_width(entry, depth + 1))
+            .max()
+            .unwrap_or_default();
+
+        let own_width = indentation(depth) + info.field.len();
+        cmp::max(own_width, max_child_width)
+    }
+
+    // Used later to align all values to the same column
+    let max_field_width = infos
+        .lscpu
+        .iter()
+        .map(|info| get_max_field_width(info, 0))
+        .max()
+        .unwrap();
```

Comment:
It's a bit confusing to have code between two functions and I would move it down.